deepchem/splits/splitters.py

```diff
@@ -1309,6 +1309,12 @@ def _generate_scaffold(smiles: str, include_chirality: bool = False) -> str:
         raise ImportError("This function requires RDKit to be installed.")
 
     mol = Chem.MolFromSmiles(smiles)
+    if mol is None:
```

Comment:
Can you add a unit test to verify this new functionality works as expected?

Reply:
added unit test